### PR TITLE
[CARBONDATA-3527] Fix 'String length cannot exceed 32000 characters' issue when load data with 'GLOBAL_SORT' from csv files which include big complex type data

### PR DESCRIPTION
**Pull request by zzcclp**
**Problem:**
When complex type data is used more than 32000 characters to indicate in csv file, and load data with 'GLOBAL_SORT' from these csv files, it will throw 'String length cannot exceed 32000 characters' exception.

**Cause:**
Use 'GLOBAL_SORT' to load data from csv files, it reads files and firstly store data in StringArrayRow, the type of all data are string, when call 'CarbonScalaUtil.getString' in 'NewRddIterator.next', it will check the length of all data and throw 'String length cannot exceed 32000 characters' exception even if it's complex type data which store as more than 32000 characters in csv files.

**Solution:**
In 'FieldConverter.objectToString' (called in 'CarbonScalaUtil.getString'), if the data type of field is complex type, don't check the length.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?  No
 
 - [ ] Any backward compatibility impacted?  No
 
 - [ ] Document update required?  No

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/570/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/571/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/575/

**Comment by zzcclp** [comment]
@jackylk @QiangCai please help to review

File: integration/spark-common-test/src/test/resources/complexdata3.csv [hunk 1]
```diff
@@ -0,0 +1,10 @@
...9\002-0.08333098\002-0.02041195\002-0.00949564\002-0.05112002\0020.01659544\0020.03327913\0020.09482127\002-0.06044888\002-0.05659582\0020.04678394\0020.04948445\0020.10130958\002-0.08592258\0020.07168256\002-0.08025738\0020.00015524\002-0.02130536\0020.05684212\0020.01048247\0020.06762113\002-0.03350449\0020.03583998\0020.05973272\002-0.03391529\0020.05963682\0020.03090304\0020.03882061\002-0.02553690\002-0.02577689\002-0.03060498\0020.01328334\002-0.01103209\0020.02207503\0020.01225353\0020.00703252\0020.00646253\0020.02086934\002-0.02023345\002-0.01639537\0020.02885322\0020.00671356\002-0.04074111\002-0.03367261\002-0.05221248\0020.01205445\002-0.08979436\0020.01642515\002-0.01522957\002-0.04506498\002-0.03454061\0020.01824338\002-0.07874620\002-0.02800509\0020.05388153\0020.08159401\002-0.03434534\0020.02791980\002-0.02704952\0020.03109635\0020.02746085\002-0.04962539\0020.01737652\002-0.04305686\002-0.02102488\0020.00829710\002-0.03093663\002-0.00534751\0020.04145030\0020.04292577\0020.00091337\002-0.01546883\0020.00670155\0020.06895591\0020.03192564\0020.03259218\002-0.01477202\0020.05259634\002-0.01358377\002-0.04051985\002-0.00872288\002-0.05790679\002-0.05036638\002-0.0036
```
Comment (jackylk) [inline integration/spark-common-test/src/test/resources/complexdata3.csv]: It is an empty file?

File: streaming/src/main/scala/org/apache/carbondata/streaming/parser/FieldConverter.scala [hunk 2]
```diff
@@ -69,7 +70,7 @@ object FieldConverter {
           val builder = new StringBuilder()
           s.foreach { x =>
             builder.append(objectToString(x, serializationNullFormat, complexDelimiters,
-              timeStampFormat, dateFormat, isVarcharType, level + 1))
+              timeStampFormat, dateFormat, isVarcharType, level = level + 1))
```
Comment (jackylk) [inline streaming/src/main/scala/org/apache/carbondata/streaming/parser/FieldConverter.scala]: It is better to move the assignment before this statement.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/578/

File: integration/spark-common-test/src/test/resources/complexdata3.csv (hunk 1 above)
Comment (zzcclp) [reply integration/spark-common-test/src/test/resources/complexdata3.csv]: > It is an empty file?

No, it's a big file and about 960K.

File: streaming/src/main/scala/org/apache/carbondata/streaming/parser/FieldConverter.scala (hunk 2 above)
Comment (zzcclp) [reply streaming/src/main/scala/org/apache/carbondata/streaming/parser/FieldConverter.scala]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/580/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/583/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/585/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/581/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/583/

**Comment by zzcclp** [comment]
@jackylk This pr is ready, please review, thanks.

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**